grafana/grafana
# Prometheus: Update lezer-promql package

**What is this feature?**

Update `lezer-promql` package to the latest version only for `grafana-prometheus`. 

`@prometheus-io/lezer-promql` 0.40.0 Changes: https://github.com/prometheus/prometheus/pull/11333

Biggest change was creating less nodes while parsing the query. It was explained [here](https://github.com/prometheus/prometheus/pull/11333) and thanks to @gtk-grafana we can [visualize](https://github.com/grafana/lezer-promql-tree-viz/) it here:
 
```
Used query: rate(process_cpu_seconds_total[1m])
```

The node tree with old version `0.37.0-rc.1`:
![image](https://github.com/grafana/grafana/assets/820946/c0638555-c419-44cc-920d-b1d8680cc8a3)

The node tree with the latest version `0.51.2`:
![image](https://github.com/grafana/grafana/assets/820946/c6cbdd73-47dd-4f3b-a8b9-ced6325a6d7d)



**Which issue(s) does this PR fix?**:

Fixes https://github.com/grafana/grafana/issues/85941

**Special notes for your reviewer:**

### How to test
- Add a promql query in code mode
  - check if it is giving you error in the right place (like pointing missing label or syntax etc)
- Switch to builder mode
  - check if you have your code visualized properly
    - there are some false negative warnings I will handle them later in a separate PR
- Create a query in builder mode and switch to the code mode
  - check if the query generated correctly.
- Disclaimer: There are some cases we weren't supported even before this upgrade. You might hit one of them. Please share those queries with me and I'll make sure they'll be fine in a separate PR

Please check that:
